packages/syft/src/syft/service/enclave/enclave_service.py

```diff
@@ -0,0 +1,209 @@
+
+    # serde / storage rules
+    __attr_searchable__ = []
+    __attr_unique__ = ["id"]
+
+    def __repr__(self) -> str:
+        return self.base_dict.__repr__()
+
+
+# TODO 🟣 Created a generic Enclave Service
+# Currently it mainly works only for Azure
+@serializable()
+class EnclaveService(AbstractService):
+    store: DocumentStore
+    service_name: Optional[str]
+
+    def __init__(self, store: DocumentStore) -> None:
+        self.store = store
+
+    @service_method(
+        path="enclave.send_user_code_inputs_to_enclave",
+        name="send_user_code_inputs_to_enclave",
+        roles=GUEST_ROLE_LEVEL,
+    )
+    def send_user_code_inputs_to_enclave(
+        self,
+        context: AuthedServiceContext,
+        user_code_id: UID,
+        inputs: Dict,
+        node_name: str,
+    ) -> Result[Ok, Err]:
+        if not context.node or not context.node.signing_key:
+            return Err(f"{type(context)} has no node")
+
+        user_code_service = context.node.get_service("usercodeservice")
+        action_service = context.node.get_service("actionservice")
+        user_code = user_code_service.stash.get_by_uid(
+            context.node.signing_key.verify_key, uid=user_code_id
+        )
+        if user_code.is_err():
+            return SyftError(
+                message=f"Unable to find {user_code_id} in {type(user_code_service)}"
+            )
+        user_code = user_code.ok()
+
+        res = user_code.status.mutate(
+            value=UserCodeStatus.EXECUTE,
+            node_name=node_name,
+            verify_key=context.credentials,
+        )
+        if res.is_err():
+            return res
+        user_code.status = res.ok()
+
+        root_context = AuthedServiceContext(credentials=context.node.verify_key)
+        user_code_update = user_code_service.update_code_state(
+            context=root_context, code_item=user_code
+        )
+        if isinstance(user_code_update, SyftError):
+            return user_code_update
+
+        if not action_service.exists(context=context, obj_id=user_code_id):
+            dict_object = DictObject(id=user_code_id)
+            dict_object.base_dict[str(context.credentials)] = inputs
+            action_service.store.set(
+                uid=user_code_id,
+                credentials=context.node.verify_key,
+                syft_object=dict_object,
+                has_result_read_permission=True,
+            )
+
+        else:
+            res = action_service.store.get(
+                uid=user_code_id, credentials=context.node.verify_key
+            )
+            if res.is_ok():
+                dict_object = res.ok()
+                dict_object.base_dict[str(context.credentials)] = inputs
+                action_service.store.set(
+                    uid=user_code_id,
+                    credentials=context.node.verify_key,
+                    syft_object=dict_object,
+                )
+            else:
+                return res
```

Comment:
return SyftError(message=res.err())

Reply:
Resolved in [ec86682](https://github.com/OpenMined/PySyft/pull/7753/commits/ec86682d4a9ca1a9241de783c8221702546275e5)